### PR TITLE
Remove ServiceAccountPermission named tuple

### PR DESCRIPTION
**Pull request by rra**
Replace it with ServiceAccountPermissionGrant for individual service
accounts and PermissionGrant when retrieving all grants for the graph.

This required adding granted_on and mapping_id to
ServiceAccountPermissionGrant.  Do the same with GroupPermissionGrant,
since this will eventually be required as well.

### CODE REVIEW

File: grouper/service_account.py [hunk 1]
```diff
@@ -199,51 +196,60 @@ def enable_service_account(session, actor, service_account, owner):
...
-    for permission in permissions:
+    for grant in grants:
         out.append(
-            ServiceAccountPermission(
-                permission=permission[0].name,
-                argument=permission[1].argument,
-                granted_on=permission[1].granted_on,
-                mapping_id=permission[1].id,
+            ServiceAccountPermissionGrant(
+                service_account=service_account,
```
Comment (wrodriguez-dbx) [inline grouper/service_account.py]: Is there a way this is automatically getting the username from `service_account`, or should it be `service_account.user.username`? If the latter I don't see how mypy isn't yelling unless it calls [`__repr__`](https://github.com/dropbox/merou/blob/4f986f60c7ce8c00b5717a890f3890eeb4b5c087/grouper/models/service_account.py#L43)

File: grouper/service_account.py [hunk 2]
```diff
@@ -199,51 +196,60 @@ def enable_service_account(session, actor, service_account, owner):
...
+                granted_on=grant.granted_on,
+                mapping_id=grant.id,
             )
         )
     return out
 
 
 def all_service_account_permissions(session):
-    # type: (Session) -> Dict[str, List[ServiceAccountPermission]]
+    # type: (Session) -> Dict[str, List[PermissionGrant]]
```
Comment (wrodriguez-dbx) [inline grouper/service_account.py]: Hmm, I'm not clear on when `PermissionGrant` is appropriate vs. `ServiceAccountPermissionGrant`. Feels like there ought to be some structuring that avoids awkardness like `service_account_permissions` returning a different type than `all_service_account_permissions`, and like multiple functions wanting to return `PermissionGrant` so filling in `is_alias` with a dummy false

File: grouper/service_account.py (hunk 1 above)
Comment (rra) [reply grouper/service_account.py]: Oh, good catch! And I think the SQLAlchemy model objects aren't fully typed; I've seen it miss other things like that.

File: grouper/service_account.py (hunk 2 above)
Comment (rra) [reply grouper/service_account.py]: Yeah, you're right, this is ugly. This made me finally (after a couple of tries) realize that `PermissionGrant` is indeed superfluous and we can use `GroupPermissionGrant` and `ServiceAccountPermissionGrant` everywhere without too many difficulties. I'll push another patch on this PR that does that.

I think the `is_alias` information is real -- it's just that we have a lot of code that isn't alias-aware yet, so there are no paths to add the aliases. But over time that attribute will become more useful.

**Pull request merged**